duckduckgo/Android
# Release Dax onboarding to all users

Task/Issue URL: https://app.asana.com/0/488551667048375/1161307587662218/f
Tech Design URL: https://app.asana.com/0/481882893211075/1168175612018809/f
CC: 

**Description**:
After releasing two different experiments to improve retention on our concept test. We are releasing the winner variant: Concept test + default browser cta during onboarding + search widget cta on home tab

Additionally, we are also releasing the concept test to all users, so we are removing all the code that will not be used from previous variants.

**Major introduced changes:**

- Database migration introducing a new table userStage to control different stages on our app. Currently: New, Dax Onboarding, Established
- Removed logic related to onboarding continue button text last screen: This logic was not used on the concept test, every screen was setting the text for that button independently to previous configurations.
- OnboardingStore usages to control onboarding replaced by UserStageStore. OnboardingStore only used to manage dax dialogs history.
- Continue Screen (Onboarding, DDG being default browser) removed. 
- EU_VARIANT: Remove feature SuppressHomeTabWidgetCta. Cta will not be shown because widget detected as installed in homescreen.
- Removed all references to conceptTest variants and all screens/resources not related to concept test.
- Adding translations for all languages

**Steps to test this PR**:

> Database migrations

Test database migration
1. Switch to develop
1. Hardcode a variant without concept test (like `sc`)
1. Fresh install
1. Navigate to browser screen
1. Switch to this branch
1. Run the app
1. Ensure browser is opened: Onboarding or Dax dialogs don't appear

Test database migration
1. Switch to develop
1. Hardcode a variant with concept test (like `mj`)
1. Fresh install
1. Navigate to browser screen (no need to visit all dax dialogs)
1. Switch to this branch
1. Run the app
1. Ensure browser is opened: Onboarding or Dax dialogs don't appear

Test database migration
1. Switch to develop
1. Hardcode a variant without concept test (like `sc`)
1. Fresh install
1. Do not perform onboarding
1. Switch to this branch
1. Run the app
1. Ensure onboarding is shown

Test database migration
1. Switch to develop
1. Hardcode a variant with concept test (like `mj`)
1. Fresh install
1. Do not perform onboarding
1. Switch to this branch
1. Run the app
1. Ensure onboarding is shown

Test database migration
1. Switch to develop
1. Fresh install
1. Add the widget to home screen
1. Open app via widget
1. Switch to this branch
1. Run the app
1. Ensure onboarding appears

Test database migration
1. Switch to develop
1. Fresh install
1. Add the widget to home screen
1. Open app via widget
1. Dismiss onboarding promo, or perform a search and navigate to browser
1. Switch to this branch
1. Run the app
1. Ensure browser is opened: Onboarding or dax dialogs don't appear

> Dialogs User flow

1. Perform fresh install
2. Ensure new onboarding is visible
3. Ensure default browser screen appears (if device compatible)
4. Continue to browser
5. Ensure Initial Dax dialog appears
6. Visit a web with trackers (cnn.com)
7. Ensure Dax Tracker dialog appears
8. Open new tab
9. Ensure End journey dialog appears
10. Perform a search on DDG
11. Ensure SERP dax dialog appears
12. Open new tab
13. Ensure search widget appears (if device compatible, if not go to step 15)
14. Dismiss widget
15. Ensure Covid-19 appears

> Additional tests:
- Try different languages like Spanish or others, to see that UI / emojis / format are not broken (at least try a few of the most used languages between our users).
---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
